lib/rpmi_hsm.c

```diff
@@ -207,7 +207,7 @@ static void __rpmi_hsm_process_hart_state_changes(struct rpmi_hsm *hsm,
 		return;
 
 	hw_state = hsm->leaf.ops->hart_get_hw_state(hsm->leaf.ops_priv, hart_index);
-	if (hart->state < 0) {
+	if ((rpmi_uint32_t)hart->state < 0) {
```

Comment:
Why this change is required ?

Reply:
hart->state is enum, that being 0xFFFFFFFF(-1), this statement will never succeed. while being  -1.

Reply:
hart->state being enum is treated as signed integer so "hart->start < 0" is true when hart->state = -1

Reply:
but in qemu its not behaving so. And one of [SO answer quoting ISO/IEC](https://stackoverflow.com/questions/159034/are-c-enums-signed-or-unsigned/159308#159308) also mentioning it could ?

Reply:
If thats the case then use rpmi_int32_t instead of rpmi_uint32_t

Reply:
my bad, its a typo.